### DISCUSSION TITLE
How can you audit GitHub OAuth permissions granted?

### DISCUSSION THREAD

**Discussion by adam-p** [community/community#21986] [New to GitHub]
<p>In light of the <a href="https://github.blog/2022-04-15-security-alert-stolen-oauth-user-tokens/" rel="noopener nofollow ugc">recent breaches</a> of the GitHub OAuth tokens held by Travis-CI and Heroku, it seems prudent to review which services you’ve given GitHub OAuth access to, <em>and what level of access</em>.</p>
<p>Seeing which services have been granted some degree of access <a href="https://github.com/settings/applications" rel="noopener nofollow ugc">is easy</a>. But I cannot see any way of discovering what level of access those services have – and knowing this is really important. Advent of Code asks for read-only access to public information, which isn’t concerning. But something like Travis-CI asks to be able to read private repos and for write access to commit status, webhooks, etc. It’s an important difference.</p>
<p>So how can we see what the access level is for these services/applications?</p>
<p>Bonus: And is there a way to review that for organization members?</p>
<p>(I originally posted this to <a href="https://stackoverflow.com/questions/71926694/how-can-you-audit-github-oauth-permissions-granted" rel="noopener nofollow ugc">SO</a> but got no traction.)</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="adam-p" data-post="1" data-topic="246191">
<div class="title">
<div class="quote-controls"></div>
 adam-p:</div>
<blockquote>
<p>Seeing which services have been granted some degree of access <a href="https://github.com/settings/applications">is easy</a>. But I cannot see any way of discovering what level of access those services have – and knowing this is really important.</p>
</blockquote>
</aside>
<p>You should see which permissions each application has if you click on the entry. Does that not work for you?</p>

**Comment by adam-p** [comment]
<p>Well… I… totally failed to click on the entries.</p>
<p>Thanks!</p>
